scripts/src/lib/lib.rs

```diff
@@ -0,0 +1,42 @@
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct Level {
+    pub title: String,
+    pub branch: String,
+    pub solution_checker: String,
+    pub flags: Vec<String>,
+}
+
+impl fmt::Display for Level {
+    // This trait requires `fmt` with this exact signature.
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // Write strictly the first element into the supplied output
+        // stream: `f`. Returns `fmt::Result` which indicates whether the
+        // operation succeeded or failed. Note that `write!` uses syntax which
+        // is very similar to `println!`.
+        write!(f, "{}", self.title)
+    }
+}
```

Comment:
Why is this function needed?

Reply:
Template rendering calls the Display impl for Level to create the node labels in the graph (which should be level titles)